sorter.py

```diff
@@ -9,23 +9,28 @@ def decode_json():
         values = json.loads(content)
     return values["Vmax"], values["lambda"]
 
-def create_weight_matrix(ranking, m):
+def sort_ranking_by_activities(ranking):
+    n,m = ranking.shape
```

Comment:
In my opinion this line raises a question. If $(n, m)$ are the dimensions of the matrix `ranking`, then it means it has been padded already. However, if I recall correctly, we discussed that the matrix `ranking_ordered` would be padded, not the first one.  
Indeed, take this example: 
```
ranking = [
[2, 1],
[3],
[1, 2, 3]
]
``` 
I believe your code won't work in this case. 

**Solution ?** Maybe we can pad in a first place the `ranking` matrix with a negative value and then handle this value in the for-loops by replacing it by  m (true after padding). As a result, the padding for the `ranking_ordered` matrix would take place in this function as well !

Reply:
Indeed we need to handle padding in those cases
Maybe to make it more elegant, when we pad we change the value of a boolean like `IS_PADDED=True`, that way we won't need to search for the padded value in `ranking_ordered` if it wasn't padded in the first place

I think the code won't work in case of padding without this previous test, but I'll test it just in case :shrug:

Reply:
Well, I think it would be clearer if the padding takes effect only on the `ranking_order` matrix. The function must be modified to achieve this. I will let you write a new version of the function.